synapseutils/sync.py

```diff
@@ -975,3 +975,49 @@ def _check_size_each_file(df):
             single_file_size = os.stat(os.path.expandvars(os.path.expanduser(file_path))).st_size
             if single_file_size == 0:
                 raise ValueError("File {} is empty, empty files cannot be uploaded to Synapse".format(file_name))
+
+
+def generateUploadManifest(syn, path, parentid, manifest_path="-"):
+    """Generate manifest for syncToSynapse from a local path."""
+    manifest_cols = ["path", "parent"]
+    manifest_rows = _walk_directory_tree(syn, path, parentid)
+    _write_manifest_data(manifest_path, manifest_cols, manifest_rows)
+
+
+def _create_folder(syn, name, parentid):
```

Comment:
This could be a set of interesting helper functions to have within synapseutils - then `synapseformation` could potentially use these helper functions.

Reply:
I'm open to this idea!